src/mantle/testkit/class-application.php

```diff
@@ -0,0 +1,501 @@
+	}
+
+	/**
+	 * Set the base path of the application.
+	 *
+	 * @param string $path Path to set.
+	 * @return static
+	 */
+	public function set_base_path( string $path ) {
+		$this->base_path = $path;
+
+		$this->instance( 'path', $this->get_base_path() );
+		$this->instance( 'path.bootstrap', $this->get_bootstrap_path() );
+		$this->instance( 'path.storage', $this->get_storage_path() );
+
+		return $this;
+	}
+
+	/**
+	 * Getter for the base path.
+	 *
+	 * @param string $path Path to append.
+	 * @return string
+	 */
+	public function get_base_path( string $path = '' ): string {
+		return $this->base_path . ( $path ? '/' . $path : '' );
+	}
+
+	/**
+	 * Get the path to the application "app" directory.
+	 *
+	 * @param string $path Path to append, optional.
+	 * @return string
+	 */
+	public function get_app_path( string $path = '' ): string {
+		$app_path = $this->app_path ?: $this->get_base_path( 'app' );
+
+		return $app_path . ( $path ? DIRECTORY_SEPARATOR . $path : $path );
+	}
+
+	/**
+	 * Set the application directory.
+	 *
+	 * @param string $path Path to use.
+	 * @return static
+	 */
+	public function set_app_path( string $path ) {
+		$this->app_path = $path;
+
+		$this->instance( 'path', $path );
+
+		return $this;
+	}
+
+	/**
+	 * Getter for the bootstrap path.
+	 *
+	 * @param string $path Path to append.
+	 * @return string
+	 */
+	public function get_bootstrap_path( string $path = '' ): string {
+		return ( $this->bootstrap_path ?: $this->base_path . DIRECTORY_SEPARATOR . 'bootstrap' ) . $path;
+	}
+
+	/**
+	 * Getter for the storage path.
+	 *
+	 * @param string $path Path to append.
+	 * @return string
+	 */
+	public function get_storage_path( string $path = '' ): string {
+		return ( $this->storage_path ?: $this->base_path . DIRECTORY_SEPARATOR . 'storage' ) . $path;
+	}
+
+	/**
+	 * Set the root URL of the application.
+	 *
+	 * @param string $url Root URL to set.
+	 */
+	public function set_root_url( string $url ) {
+		$this->root_url = $url;
+	}
+
+	/**
+	 * Getter for the root URL.
+	 *
+	 * @param string $path Path to append.
+	 * @return string
+	 */
+	public function get_root_url( string $path = '' ): string {
+		return $this->root_url . ( $path ? '/' . $path : '' );
+	}
+
+	/**
+	 * Get the cache folder root
+	 * Folder that stores all compiled server-side assets for the application.
+	 *
+	 * @return string
+	 */
+	public function get_cache_path(): string {
+		return $this->get_bootstrap_path( '/cache' );
+	}
+
+	/**
+	 * Get the cached Composer packages path.
+	 *
+	 * Used to store all auto-loaded packages that are Composer dependencies.
+	 *
+	 * @return string
+	 */
+	public function get_cached_packages_path(): string {
+		return $this->get_cache_path() . '/packages.php';
+	}
+
+	/**
+	 * Get the cached model manifest path.
+	 * Used to store all auto-registered models that are in the application.
+	 *
+	 * @return string
+	 */
+	public function get_cached_models_path(): string {
+		return '';
```

Comment:
is this temporary? should there be a `@todo`?

Reply:
Nope, the teskit has a lot of stubbed out methods because of the interface it implements that aren't needed. So they return empty/null